pkg/typesystem/typesystem.go

```diff
@@ -709,7 +713,11 @@ func hasEntrypoints(
 				return false, false, fmt.Errorf("undefined type definition for '%s#%s'", assignableTypeName, assignableRelationName)
 			}
 
-			if _, ok := v[assignableTypeName][assignableRelationName]; ok {
+			if hasEntrypoint, ok := v[assignableTypeName][assignableRelationName]; ok {
+				if hasEntrypoint {
+					return true, false, nil
```

Comment:
i think these lines should stay (to be consistent with line 768), but i can't think of a test that covers this case :(